executor/source_task_executors/gcm_task_executor.py

```diff
@@ -0,0 +1,114 @@
+import json
+
+from google.protobuf.wrappers_pb2 import StringValue, UInt64Value
+
+from connectors.utils import generate_credentials_dict
+from executor.playbook_source_manager import PlaybookSourceManager
+from executor.source_processors.gcm_api_processor import GcmApiProcessor
+from protos.base_pb2 import Source, SourceModelType
+from protos.connectors.connector_pb2 import Connector as ConnectorProto
+from protos.playbooks.playbook_commons_pb2 import PlaybookTaskResult, TableResult, PlaybookTaskResultType
+from protos.playbooks.source_task_definitions.gcm_task_pb2 import Gcm
+
+
+class GcmSourceManager(PlaybookSourceManager):
+
+    def __init__(self):
+        self.source = Source.GCM
+        self.task_proto = Gcm
+        self.task_type_callable_map = {
+            Gcm.TaskType.GET_METRICS: {
+                'executor': self.get_metrics,
+                'model_types': [SourceModelType.GCM_METRIC],
+                'result_type': PlaybookTaskResultType.TABLE,
+                'display_name': 'Get Metrics from GCM',
+                'category': 'Monitoring'
+            },
+            Gcm.TaskType.GET_LOGS: {
+                'executor': self.get_logs,
+                'model_types': [SourceModelType.GCM_LOGS],
+                'result_type': PlaybookTaskResultType.TABLE,
+                'display_name': 'Get Logs from GCM',
+                'category': 'Logging'
+            },
+        }
+
+    def get_connector_processor(self, gcm_connector, **kwargs):
+        generated_credentials = generate_credentials_dict(gcm_connector.type, gcm_connector.keys)
+        return GcmApiProcessor(**generated_credentials)
+
+    def get_metrics(self, gcm_task: Gcm, gcm_connector: ConnectorProto) -> PlaybookTaskResult:
+        try:
+            if not gcm_connector:
+                raise Exception("Task execution Failed:: No GCM source found")
+
+            gcm_command = gcm_task.get_metrics
```

Comment:
If there are global variables to this command then they should be populated inside this. Add the logic for that